doc/hosts/appservice.md

```diff
@@ -0,0 +1,72 @@
+[continuous deployment]: https://docs.microsoft.com/en-us/azure/app-service/deploy-continuous-deployment?toc=%2fazure%2fapp-service%2fcontainers%2ftoc.json
+
+## Overview
+
+When an application is pushed to App Service, e.g. through local git or GitHub integration, the
+source code will be available in `/home/site/repository`. Through a hook in the git repository,
+after the code is pushed App Service will call Oryx to build the application if a deployment
+script wasn't provided. After the build step the web app is placed in `/home/site/wwwroot`,
+location from which it will be executed.
+
+App Service has two components that are relevant in this discussion: the [Kudu][] service,
+where the build happens, and the runtime environments where the web apps are executed. Both environments
+share files through the `/home` network directory.
+
+As a side note, since `/home` is a shared location for both build and runtime, and is also a 
+persistent storage location, some web apps use it to store state. We don't recomment this since
+it's easy to run into concurrent access issues. We recommend the use of services specifically
+designed for [storage][], which might also include backups, replication, and much more.
+
+[Kudu]: https://github.com/azure-App-Service/kudulite
+[storage]: https://azure.microsoft.com/en-us/product-categories/storage/
+
+## Node.js
+
+In general, node.js applications have a large number of package dependencies, either directly or indirectly,
+i.e. the dependencies of their dependencies. Since each package might contain several `.js` files, fetching 
+dependencies means a lot of disk IO operations. Since in the App Service model the application is stored in a 
+network volume, the `/home` directory, fetching and storing the packages alongside the application in 
+`/home/site/wwwroot` means a lot of IO operations would have to go through the network. 
+
+### Compressing node modules
+
+Recently we've made some build performance improvements that made the builds run up to ten times faster. We achieved
+this by fetching the packages outside of the network location, compress its contents, and maintain all the dependencies
+in a single file, `node_modules.tar.gz`, that is located in `/home/site/wwwroot`. As a result, we no longer maintain the
+`node_modules` folder in this `/home/site/wwwroot`.
+
+As part of the application startup process, we extract the contents of `node_modules.tar.gz` to `/node_modules`, which
+is outside of the volume share. Since the node runtime looks for packages inside directories called `node_modules`
+starting at the application directory (`/home/site/wwwroot`) all the way to `/`, it is able to find the extracted
+packages.
+
+If for some reason you want to disable this behavior, e.g. you have hardcoded references to files inside 
+`node_modules`, you can set the app setting `BUILD_FLAGS` to `Off`. Note that casing matters, so `off` won't work.
```

Comment:
Interesting, so `BUILD_FLAGS` is the flag that AppService chose to indicate this? how clear :-)